shamil_latypov/logger.cpp

```diff
@@ -0,0 +1,34 @@
+#include "logger.hpp"
+#include <fstream>
+#include <iostream>
+#include <optional>
+#include <stdexcept>
+
+namespace uni_cource_cpp {
+
+void Logger::set_output_file_path(const std::optional<std::string>& file_path) {
+  if (file_path == std::nullopt) {
+    if (output_stream_.has_value()) {
+      output_stream_.value().close();
+    }
```

Comment:
Зачем нам продолжать выполнение?
Закрывается аавтоматически при удалении
```suggestion
    output_stream_ = std::nullopt;
    return;
```